modules/gdscript/tests/scripts/runtime/features/anonymous_extend_user.gd

```diff
@@ -0,0 +1,5 @@
+extends "/Users/dennis.ploeger/code/private/godot/godot/modules/gdscript/tests/scripts/runtime/features/anonymous_extend.gd"
```

Comment:
As a unit test, this should be modified so that it is not specific to your machine, but relative to the Godot install directory. I forget how to access that, but it's in one of the singletons (maybe Engine?).

Reply:
Why can't you use function calls in the extends statement? Is it not possible to wrap the StringName in a String? Maybe I'm just out of the context.

Reply:
Maybe I didn't understand you there. I thought you meant I should use a function from a singleton to get the relative Godot install directory in the "extends" keyword in this GDScript file. That's not possible. Extends only accepts strings.

Reply:
I meant, in the `GDScriptLanguage::get_global_class_name`, when it is *evaluating* the `extends` value(s) provided to the GDScript file, you can, "use a function from a singleton to get the relative Godot install directory"

Reply:
Oh okay, you meant fixing the relative path issue. I understood you wanted to add a new ticket about it because it was out of scope.

I'll look into it then.

Reply:
Oh, I must've misunderstood something you mentioned. Had it in my mind you were going to attempt it. Either way, up to you. Certainly wouldn't hurt to have it.

Reply:
There you go. I optimized the code in get_global_class_name as the todo there suggested and used the gdscript analyzer. That also solved the relative extend problem (which was only a problem for that specific method because of its own way of finding the base class).

Please check my code there. It works with the test suite, but I'm not that deep in the engine code to know if it makes sense the way I coded it (and my C++ is definitely rusty 😆 )

Reply:
Got it. Thanks so much! I'll take a look as soon as I can find the time. Have a lot of things that need to be done by Monday though, so bear with me.

Reply:
Hey, Just wanted to update you. I've been sidetracked at work and haven't yet been able to complete the big feature I've been working on (it's been, like, a 3-month WIP thus far), but I need to finish it ASAP, so that's why I've been out of contact for so long. But I promise that as soon as I get an opening, I'll review your work and merge as soon as I can. Thank you *so* much!